keras_nlp/samplers/sampler.py

```diff
@@ -230,10 +231,12 @@ def __call__(
         # , and turn on `jit_compile` accordingly.
         sample = self.sample
         if not self.run_eagerly:
-            sample = tf.function(self.sample, jit_compile=self.jit_compile)
+            if getattr(self, "jit_compile", True):
+                sample = self._sample_graph
```

Comment:
why not just inline all of this

`sampler = tf.function(self.sample, jit_compile=self.jit_compile)` would save a bit

also, same comment here as above. set `jit_compile = True` (or False) in init, avoid getattr

Reply:
`sample = tf.function(self.sample, jit_compile=self.jit_compile)` still causes recompile if the call is under a for-loop, my guess is it is considered as "defining tf.function in a for-loop". 

Fixed the getattr stuff.